src/EdFi.AnalyticsMiddleTier.DataStandard31/Asmt/MSSQL/0014-View-StudentAssessmentFact-Create.sql

```diff
@@ -0,0 +1,118 @@
+﻿-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+IF EXISTS (SELECT 1 FROM INFORMATION_SCHEMA.VIEWS WHERE TABLE_SCHEMA = 'analytics' AND TABLE_NAME = 'asmt_StudentAssessmentFact')
+BEGIN
+	DROP VIEW analytics.asmt_StudentAssessmentFact
+END
+GO
+
+CREATE VIEW analytics.asmt_StudentAssessmentFact AS
+
+	SELECT
+        CONCAT(StudentAssessment.AssessmentIdentifier, '-', StudentAssessment.Namespace, '-', StudentAssessment.StudentAssessmentIdentifier, '-', StudentAssessment.StudentUSI) AS StudentAssessmentKey,
+		CASE WHEN SASOA.StudentUSI IS NULL THEN NULL ELSE CONCAT(SASOA.StudentUSI, '-', SASOA.IdentificationCode, '-', SASOA.AssessmentIdentifier, '-', SASOA.StudentAssessmentIdentifier, '-', SASOA.Namespace) END AS StudentObjectiveAssessmentKey,
+        CASE WHEN SASOA.AssessmentIdentifier IS NULL THEN NULL ELSE CONCAT(SASOA.AssessmentIdentifier, '-', SASOA.IdentificationCode, '-', SASOA.Namespace) END AS ObjectiveAssessmentKey,
+		CONCAT(Assessment.AssessmentIdentifier, '-', StudentAssessment.Namespace) AS AssessmentKey,
+        Assessment.AssessmentIdentifier,
+        StudentAssessment.Namespace,
+        StudentAssessment.StudentAssessmentIdentifier,
+        StudentAssessment.StudentUSI,
+        CONCAT(Student.StudentUniqueId, '-', StudentSchoolAssociation.SchoolId) AS StudentSchoolKey,
+        School.SchoolId AS SchoolKey,
+        CONVERT(varchar, StudentAssessment.AdministrationDate, 112) as AdministrationDate,
+        COALESCE(ISNULL(StudentAssessmentScoreResult.Result,SASOASR.Result),'') AS StudentScore,
```

Comment:
Why do we have a ISNULL inside a Coalesce? Maybe we can remove the ISNULL statement. Unless I am not seeing something.

Reply:
resolved, please review it again.